impeller/display_list/dl_dispatcher.cc

```diff
@@ -484,12 +488,24 @@ static std::shared_ptr<ColorFilter> ToColorFilter(
 // |flutter::DlOpReceiver|
 void DlDispatcher::setColorFilter(const flutter::DlColorFilter* filter) {
   // Needs https://github.com/flutter/flutter/issues/95434
-  paint_.color_filter = ToColorFilter(filter);
+  if (paint_.color_filter) {
+    auto color_filter = ToColorFilter(filter);
+    paint_.color_filter =
+        ColorFilter::MakeComposed(paint_.color_filter, color_filter);
```

Comment:
Does the paint state get cleared out between every draw call? I think this solution implicitly relies on `setColorFilter` and `setInvertColors` always being called exactly once per dispatch in this solution.

Reply:
ugh yeah

Reply:
I hate this

Reply:
reverted!